network/f5/bigip_gtm_virtual_server.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2015, Michael Perzel
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: bigip_gtm_virtual_server
+short_description: "Manages F5 BIG-IP GTM virtual servers"
+description:
+    - "Manages F5 BIG-IP GTM virtual servers"
+version_added: "2.0"
+author: 'Michael Perzel'
+notes:
+    - "Requires BIG-IP software version >= 11.4"
+    - "F5 developed module 'bigsuds' required (see http://devcentral.f5.com)"
+    - "Best run as a local_action in your playbook"
+    - "Tested with manager and above account privilege level"
+
+requirements:
+    - bigsuds
+options:
+    server:
+        description:
+            - BIG-IP host
+        required: true
+    user:
+        description:
+            - BIG-IP username
+        required: true
+    password:
+        description:
+            - BIG-IP password
+        required: true
+    state:
+        description:
+            - Virtual server state
+        required: true
+        choices: ['present', 'absent','enabled','disabled']
+    virtual_server_name:
+        description:
+            - Virtual server name
+        required: True
+    virtual_server_server:
+        description:
+            - Virtual server server
+        required: true
+    host:
+        description:
+            - Virtual server host
+        required: false
```

Comment:
if you an arg to required: false, a default must be specified even the default is null/None